### PR TITLE
chore(ARCH-482/ListView): remove deprecated lifecycle and use RTL for test

### PR DESCRIPTION
**Pull request by jsomsanith-tlnd**
**What is the problem this PR is trying to solve?**
Deprecated lifecycle

**What is the chosen solution to this problem?**
Remove deprecated lifecycle and migrate tests to RTL

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/Checkbox/Checkbox.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ Checkbox.defaultProps = {
 
 Checkbox.propTypes = {
 	id: PropTypes.string.isRequired,
-	label: PropTypes.string,
+	label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
```
Comment (Benrajalu) [inline packages/components/src/Checkbox/Checkbox.js:55]: That's not just using RTL and changing lifecycles, that's an large API change 🚓

File: packages/forms/src/UIForm/fields/ListView/ListView.component.js [hunk 2]
```diff
@@ -49,16 +49,16 @@ class ListViewWidget extends React.Component {
...
+		const { schema: prevSchema, value: prevValue } = prevProps;
+		const { schema: nextSchema, value: nextValue } = this.props;
+		if (prevSchema !== nextSchema) {
 			this.setState(oldState =>
-				initItems(schema, value, oldState.searchCriteria, this.onToggleItem.bind(this)),
+				initItems(nextSchema, nextValue, oldState.searchCriteria, this.onToggleItem.bind(this)),
 			);
-		} else if (value !== this.props.value) {
-			this.setState(oldState => updateItems(oldState.items, value, oldState.searchCriteria));
+		} else if (prevValue !== nextValue) {
```
Comment (Benrajalu) [inline packages/forms/src/UIForm/fields/ListView/ListView.component.js:60]:
```suggestion
		} else if (prevValue !== nextValue) {
```
You're only comparing object instances though. If the values are the same, are you ok with landing there?

File: packages/forms/src/UIForm/fields/ListView/ListView.component.test.js [hunk 3]
```diff
@@ -66,151 +58,160 @@ describe('ListView field', () => {
...
-			expect(wrapper.find('input[id^="checkbox-my-list-view-"]').length).toBe(2);
-			expect(wrapper.find('.theme-tc-item-container label').at(1).text()).toBe('Albania');
-			expect(wrapper.find('.theme-tc-item-container label').at(2).text()).toBe('Algeria');
+			expect(screen.getAllByRole('option').length).toBe(2);
+			expect(screen.getByRole('checkbox', { name: 'Deselect Albania' })).toBeInTheDocument();
+			expect(screen.getByRole('checkbox', { name: 'Deselect Algeria' })).toBeInTheDocument();
 		});
 
-		it('should show no result message', () => {
+		it('should display a message when no results was found', () => {
```
Comment (Benrajalu) [inline packages/forms/src/UIForm/fields/ListView/ListView.component.test.js:157]:
```suggestion
		it('should display a message when no results was found', () => {
```
```suggestion
		it('should display a message when no results were found', () => {
```

**Review by Benrajalu — APPROVED** [review]

**Review by romainseb — APPROVED** [review]

File: packages/components/src/Checkbox/Checkbox.js (hunk 1 above)
Comment (jmfrancois) [reply packages/components/src/Checkbox/Checkbox.js:55]: Definition of node:
/ Anything that can be rendered: numbers, strings, elements or an array so you can just replace it by node

**Review by jmfrancois — APPROVED** [review]
with suggestion

File: packages/components/src/Checkbox/Checkbox.js (hunk 1 above)
Comment (jsomsanith-tlnd) [reply packages/components/src/Checkbox/Checkbox.js:55]: The propTypes were not correct, the component was already used with components. This is to remove the proptypes warnings

Comment (jsomsanith-tlnd) [reply packages/components/src/Checkbox/Checkbox.js]: Merged your suggestion JM

File: packages/forms/src/UIForm/fields/ListView/ListView.component.js (hunk 2 above)
Comment (jsomsanith-tlnd) [reply packages/forms/src/UIForm/fields/ListView/ListView.component.js:60]: Yeah,the PR is not about rethinking the component lifecycle, it's a migration with exact same feature

File: packages/forms/src/UIForm/fields/ListView/ListView.component.test.js (hunk 3 above)
Comment (jsomsanith-tlnd) [reply packages/forms/src/UIForm/fields/ListView/ListView.component.test.js]: Thanks for the fix

**Pull request merged**